pkg/yurthub/healthchecker/health_checker.go

```diff
@@ -181,50 +190,66 @@ func newChecker(
 
 	nl := NewNodeLease(kubeClient, cfg.NodeName, defaultLeaseDurationSeconds, cfg.HeartbeatFailedRetry)
 	c := &checker{
-		nodeLease:        nl,
-		lastTime:         time.Now(),
-		clusterHealthy:   false,
-		healthyThreshold: cfg.HeartbeatHealthyThreshold,
-		healthyCnt:       0,
-		remoteServer:     url,
-		onFailureFunc:    tp.Close,
-		setLastNodeLease: setLastNodeLease,
-		getLastNodeLease: getLastNodeLease,
+		nodeLease:              nl,
+		lastTime:               time.Now(),
+		clusterHealthy:         false,
+		healthyThreshold:       cfg.HeartbeatHealthyThreshold,
+		healthyCnt:             0,
+		remoteServer:           url,
+		onFailureFunc:          tp.Close,
+		setLastNodeLease:       setLastNodeLease,
+		getLastNodeLease:       getLastNodeLease,
+		healthCheckGracePeriod: cfg.KubeletHealthGracePeriod,
 	}
 	return c, nil
 }
 
 func (c *checker) check() bool {
-	baseLease := c.getLastNodeLease()
-	lease, err := c.nodeLease.Update(baseLease)
-	if err == nil {
-		if err := c.setLastNodeLease(lease); err != nil {
-			klog.Errorf("set last node lease fail: %v", err)
+	var err error
```

Comment:
`check()` is used for sending lease to remote server and checking healthy status. and we need to stop `check()` when kubelet stopped renew node lease.  so i think add `kubelet stopped renew node lease verification` in `func (hcm *healthCheckerManager) sync()` maybe more readable. like as following:

```
func (hcm *healthCheckerManager) sync() {
        // check kubelet stopped renew node lease or not
	isKubeletStoped, now := false, time.Now()
	lastKubeletLeaseReqTime := c.getLastKubeletLeaseReqTime()
	if !lastKubeletLeaseReqTime.IsZero() && c.healthCheckGracePeriod > 0 {
		isKubeletStoped = now.After(lastKubeletLeaseReqTime.Add(c.healthCheckGracePeriod))
	}

	// Ensure that the node heartbeat can be reported when there is a healthy remote server.
	//try detect all remote server in a loop, if there is an remote server can update nodeLease, exit the loop.
	for i := 0; i < len(hcm.remoteServers); i++ {
		c := hcm.getChecker()
                if isKubeletStoped {
                       klog.Infof("kubelet does not post lease request for more than %v, stop renew node lease and assume remote server %s is unhealthy", now.Sub(lastKubeletLeaseReqTime), c.remoteServer.String())
                      // do other settings.
                      continue              
                }

		if c.check() {
			break
		}
	}
}
```